source/mainnet/smart-contracts/tutorials/piggy-bank/preparing.rst

```diff
@@ -0,0 +1,256 @@
+
+The :ref:`run a node <node-requirements>` guide will help you set up your instance correctly. It is recommended to use an ubuntu instance on a server in the cloud for this tutorial.
+
+.. Note::
+   It is technically fine to run your testnet node locally on your machine instead of on a server in the cloud. Since blockchain nodes have to run 24/7 to be up-to-date with the blockchain, you have to run your local machine 24/7. Alternatively, if you don't want to run your local machine 24/7, you can let your node catch up whenever you start your machine. Because this takes some time, this tutorial recommends a cloud provider setup for convenience.
+
+Start the syncing process of the testnet node by following the guide for your platform :ref:`Ubuntu<ubuntu-node>`, :ref:`Docker<docker-node>`, :ref:`Windows<windows-node>`, or :ref:`MacOS<macos-node>`.
+
+You should find your node name on the `Concordium testnet dashboard <https://dashboard.testnet.concordium.com/>`__. It will take less than a day until your testnet node is fully synced. You can observe the syncing process by watching the finalization length of your node. Wait until the ``Fin Length`` (finalization length) of your node is the same as the highest value used by the majority of nodes.
+
+.. Note::
+   To allow the network dashboard to display nodes in the network and their current status, a node must run a collector. The collector is enabled by the default start-up script but can be disabled.
+
+.. image:: ./images/pb_tutorial_13.png
+   :width: 100 %
+
+Alternatively, you can query the syncing state of your node with the ``concordium-client`` tool. You will learn how to set up the ``concordium-client`` tool in the next section. Feel free to come back to this command after reading the next section.
+
+.. code-block:: console
+
+   ./concordium-client consensus status --grpc-port 10001
+
+
+.. Note::
+   It is a good practice to enable inbound connections on port 8889 (testnet) in your instance. You can allow inbound connections from any IPv4 and IPv6 address, by selecting ``0.0.0.0/0`` and ``::/0`` on the port 8889. This is not mandatory for the node to sync but it will make your node a good network participant. Feel free to skip this step if you are not feeling confident editing the inbound connection rules of your instance.
+
+.. image:: ./images/pb_tutorial_12.png
+   :width: 100 %
+
+.. _interacting-with-your-testnet-node:
+
+Interacting with your testnet node
+==================================
+
+You are now ready to download the :ref:`concordium-client<concordium-node-and-client-download-testnet>` package. Please rename the package to just ``concordium-client`` in case it has some version annotation so you can follow the commands in this guide easily.
+
+.. Note::
+   If you are not using ubuntu/linux as your operating system the following screenshots look differently. Please remember to adjust the following commands based on your operating system if you are not using ubuntu/linux.
```

Comment:
There is repetition in these two sentences. Please rephrase the "if you are not using ubunut/linux".